src/components/Demo/DemoCreate/DemoCreateStart/index.tsx

```diff
@@ -1,41 +1,45 @@
-import { Button, LoaderSpinner } from "@govtechsg/tradetrust-ui-components";
-import React, { FunctionComponent, useContext, useState } from "react";
+import { Button } from "@govtechsg/tradetrust-ui-components";
+import React, { FunctionComponent, useContext, useEffect, useState } from "react";
 import { ethers } from "ethers";
-import { useDispatch } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import { ProviderContext } from "../../../../common/contexts/provider";
-import { deployingDocStore } from "../../../../reducers/demo-create";
+import { deployingDocStore, getDocumentPrepared } from "../../../../reducers/demo-create";
 import { getFunds } from "../../../../services/create";
 import { DemoCreateContext } from "../contexts/DemoCreateContext";
+import { LoadingModal } from "../../LoadingModal";
 
 export const DemoCreateStart: FunctionComponent = () => {
   const { account, provider } = useContext(ProviderContext);
   const { setActiveStep } = useContext(DemoCreateContext);
   const [loading, setLoading] = useState(false);
-  const [error, setError] = useState("");
+
+  const { prepared, error } = useSelector(getDocumentPrepared);
 
   const dispatch = useDispatch();
 
   const handleStart = async () => {
     setLoading(true);
 
-    try {
-      const balance = await provider.getBalance("latest");
-      const formattedBalance = Number(ethers.utils.formatEther(balance));
+    const balance = await provider.getBalance("latest");
+    const formattedBalance = Number(ethers.utils.formatEther(balance));
 
-      if (formattedBalance <= 1) {
-        await getFunds(account as string);
-      }
+    if (formattedBalance <= 1) {
+      await getFunds(account as string);
+    }
```

Comment:
think might need to try catch here? if faucet no funds, then throw error. otherwise ui will hang at the `modal preparing`